src/components/clusters/clusterEventsModal.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { Button, Modal } from '@patternfly/react-core';
+import { ToolbarButton } from '../ui/Toolbar';
+import EventListFetch from '../fetching/EventListFetch';
+import { Cluster } from '../../api/types';
+
+type ClusterEventsModalButtonProps = React.ComponentProps<typeof Button> & {
+  ButtonComponent?: typeof Button | typeof ToolbarButton;
+  onClick?: () => void;
+  clusterId: Cluster['id'];
+};
+
+export const ClusterEventsModalButton: React.FC<ClusterEventsModalButtonProps> = ({
+  ButtonComponent = ToolbarButton,
+  onClick,
+  clusterId,
+  ...props
+}) => {
+  const [isModalOpen, setIsModalOpen] = React.useState(false);
+  const closeModal = () => setIsModalOpen(false);
+  const handleClick = onClick || (() => setIsModalOpen(true));
+  return (
+    <>
+      <ButtonComponent {...props} onClick={handleClick}>
+        View Cluster Events History
+      </ButtonComponent>
+      <ClusterEventsModal isOpen={isModalOpen} onClose={closeModal} clusterId={clusterId} />
+    </>
+  );
+};
+
+type ClusterEventsModalProps = {
+  clusterId: Cluster['id'];
+  onClose: () => void;
+  isOpen: boolean;
+};
+
+export const ClusterEventsModal: React.FC<ClusterEventsModalProps> = ({
+  clusterId,
+  onClose,
+  isOpen,
+}) => {
+  return (
+    <Modal title="Cluster Events" isOpen={isOpen} onClose={onClose} isFooterLeftAligned isLarge>
+      <EventListFetch entityId={clusterId} entityKind="cluster" />
```

Comment:
The table in modal is rendered always with vertical scrollbar, no matter the content can fit.

Reply:
Yeah, this seems like a bug in modal/table. I'd like to try to fix it as a separate issue. It is most likely PF problem

Reply:
Agree, it's PF issue. But we can workaround it in our code base, it can be fixed by playing with `height`.